templates/_components/_image.twig

```diff
@@ -493,40 +493,54 @@
     alt: imageAlt|e('html_attr'),
     width:  not imageIsResponsive ? imageDefault.width * imageScaleFactor,
     height: not imageIsResponsive ? imageDefault.height * imageScaleFactor,
-    
+
   } | merge(imageLazyload ? {
     src: craft.imager.base64Pixel(imageDefault.width, imageDefault.height, imageColour),
     'data-src': imageDefault.url,
-    
+
   }:{}) | merge(imageTransforms is defined and imageTransforms|length > 1 ? {
     sizes: imageSizesGuide ?? false,
     srcset: not imageLazyload ? defaultSrcSet : false,
     'data-sizes': imageLazyload and not (imageSizesGuide ?? true) ? 'auto',
     'data-srcset': imageLazyload ? defaultSrcSet,
   }:{})) }}
-  
+
   {% if imageSources|length %}
     </picture>
   {% endif %}
 
   {% if imageHasZoom %}
     </a>
   {% endif %}
-  
+
 {% elseif imageAsset and imageIsCss %}
+  {% set imageSetVersions = ['-webkit-image-set', 'image-set'] %}
   {% minify css %}
     <style>
       .{{ imageCssClass }} {
         background-image: url('{{ imageDefault.url }}');
       }
       {% for source in imageSources|reverse %}
       @media {{ source.media }} {
-        {{ (source.type ?? '') == 'image/webp' ? '.has-webp ' -}}
         .{{ imageCssClass }} {
-          background-image: url('{{ (source.transforms|last).url }}');
+          {% if source.type is defined %}
+            {% set fallbackImage = imageSources
+              |reverse
+              |map(i => i.type is not defined and i.media == source.media ? i.transforms|last)
+              |filter %}
+
+            {% set fallbackImageUrl = (fallbackImage|last).url ?? imageDefault.url %}
+            {% set fallbackImageType = (fallbackImage|last).mimeType ?? imageDefault.mimeType %}
+            {% for imageSet in imageSetVersions %}
+              background-image: {{ imageSet }}(
+                url('{{ (source.transforms|last).url }}') type('{{ source.type }}'),
+                url('{{ fallbackImageUrl }}') type('{{ fallbackImageType }}'));
+            {% endfor %}
+          {% else %}
+            background-image: url('{{ (source.transforms|last).url }}');
+          {% endif %}
         }
       }
-      {% endfor %}
```

Comment:
Do we not need the `{% endfor %}` tag here to close the for loop opened in line 523? Perhaps the indentation can be fixed as well to make it easier to determine this?